src/bp/ui-studio/src/web/views/OneFlow/diagram/ContentForm/TextField.tsx

```diff
@@ -42,9 +44,36 @@ const TextAreaList: FC<Props> = ({
     })
   }
 
+  const getRefLang = (value, variations, currentLang, defaultLang) => {
+    const combinedValues = Object.keys(value).reduce(
+      (acc, key) => ({ ...acc, [key]: [value?.[key] || '', ...(variations[key] || [])].filter(Boolean) }),
+      {}
+    )
+    const currentLangValues = combinedValues[currentLang]
+    let currentHighest = 0
+    let refLang
+    Object.keys(combinedValues)
+      .filter(l => l !== currentLang)
+      .forEach(key => {
+        const thisLength = combinedValues[key].length
+        if (thisLength > currentHighest) {
+          refLang = key
+          currentHighest = thisLength
+        }
+      })
+
+    if (currentLangValues?.length || 0 < currentHighest) {
+      return refLang
+    }
+  }
```

Comment:
ref lang = default lang, is this block still necessary ?

Reply:
I'm not undoing what I've done before, it's overkill, but it works so it'll be like that until we implement the translation center